content/documents/working-with-specs.mdx

```diff
@@ -0,0 +1,210 @@
+---
+type:           article
+title:          Working with Product Specs
+description:    Creating complex products with specs.
+author:         asutherland
+publishDate:    2021-12-21
+updatedDate:
+tags:           ["Products", "Variants", "Specs", "Options"]
+---
+
+## What is a Spec?
+
+While a product may have a number of properties that provide details about the product, a spec refers to a product specification that typically allows the user to either provide some text input or choose from options, which can result in customising the product and/or selecting a variation of the product.
+
+For example, when selling a pen we may request a name to be engraved on the side, providing a customised product, whereas a television may offer different sizes representing variant products.
+
+## Spec Types
+
+The configuration of a spec defines how a user will interact with a product it has been assigned to. Specs can generally be categorised under one of the following types:
+
+### Open Text Spec
+
+The `AllowOpenText` property provides the user the option of inputting text associated to the product, e.g. a name to be engraved onto a pen.
+
+<CodeExample
+  title="Creating an open text spec"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8;\n
+{
+    "ID": "OPEN_TEXT_SPEC",
+    "Name": "Open Text Spec",
+    "AllowOpenText": true
+}`
+}}
+/>
+
+### Options Spec
+
+To give the user the ability to modify a product configuration while still limiting their selection, a spec with options can be created, e.g. business cards may have a number of design templates to choose from.
+
+<CodeExample
+  title="Creating an options spec"
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/specs HTTP/1.1
```

Comment:
Given the recent introduction to [regions](https://ordercloud.io/knowledge-base/ordercloud-regions) and to account for various [environments](https://ordercloud.io/knowledge-base/ordercloud-environments) it might make sense to start parameterizing the url we display in docs so that it is `https://{region}-{environment}.ordercloud.io`

Reply:
A fair point. Currently was looking to be consistent with the Learn docs. Any chance we have any guidelines/conventions that we should adhere to for documentation? If not, we could probably looking at starting one to ensure consistency.

Reply:
That URL example won't work for the US West region, it's just `https://{environment}.ordercloud.io`. We should look into standardizing the formatting to convey the notion of regions before we change the documentation.

Reply:
Following Alexa's guidance and only updated to use environment for now.

Reply:
Please just use `sandboxapi` for the consistency of our documentation

Reply:
The following URLs all work as expected. I'm talking to Rob about updating it in portal but I think it makes sense to 

https://uswest-sandbox.ordercloud.io/env
https://uswest-staging.ordercloud.io/env

Reply:
The following URLs all work as expected. 

https://uswest-sandbox.ordercloud.io/env
https://uswest-staging.ordercloud.io/env
https://uswest-production.ordercloud.io/env

I'm talking to Rob about updating it in portal but I don't really see a reason we can't make this change ahead of that. I do think it makes sense to submit as  a PR separate from this one though, along with the guidelines/conventions doc you had brought up Andrew

Reply:
This documentation stated that they remained unchanged so I was not aware those were working URLs. This should be a separate effort to update all documentation in one pull request for consistency rather than piece by piece. https://ordercloud.io/knowledge-base/ordercloud-regions